fhir-registry/src/main/java/com/ibm/fhir/registry/resource/FHIRRegistryResource.java

```diff
@@ -111,8 +111,8 @@ public boolean isDefaultVersion() {
         return defaultVersion;
     }
 
-    public Resource getResource() {
-        return resource;
+    public <T extends Resource> T getResource() {
+        return (T) resource;
```

Comment:
didn't actually mean to include this one... @JohnTimm still think its better to force the users into an explicit cast?

Reply:
I think this change seems somewhat unnecessary. `FHIRRegistry.getResource()` already does casting and we also have the Resource.as(Class<? extends Resource>) method.

Reply:
yeah, here is the difference for users of this method.

old:
```
SearchParameter sp = registryResource.getResource().as(SearchParameter.class);
```

new:
```
SearchParameter sp = registryResource.getResource();
```

Not a huge win, but seems like a slight improvement to me.

Reply:
I backed it out since it doesn't really belong with this PR anyway

Reply:
> yeah, here is the difference for users of this method.
> 
> old:
> 
> ```
> SearchParameter sp = registryResource.getResource().as(SearchParameter.class);
> ```
> 
> new:
> 
> ```
> SearchParameter sp = registryResource.getResource();
> ```
> 
> Not a huge win, but seems like a slight improvement to me.

In most cases I think clients will use `FHIRRegistry.getInstance().getResource(url, SearchParameter.class)` and won't be working with `FHIRRegistryResource` instance directly.